nbconvert/exporters/pdf.py

```diff
@@ -172,12 +172,11 @@ def from_notebook_node(self, nb, resources=None, **kw):
             self.log.info("Building PDF")
             rc = self.run_latex(tex_file)
             if rc:
-                rc = self.run_bib(tex_file)
-            if rc:
+                self.run_bib(tex_file)
```

Comment:
The only problem that I see with removing the if statement is that it means running latex an extra 3 times even if we don't need it. Before the second latex would only run if bibtex was successful.

Reply:
@t-makaro I agree.

However, it seems that manually running xelatex on the nbconverted-converted notebook2.ipynb fails (returns non-zero exit code) though it does produce a .pdf file. To me this seems like a bad .tex file to run tests on -- so maybe the conversion from ipynb to tex is incorrect somehow? As I indicated above I think a non-zero return from xelatex should return a LatexFailed exception.

Reply:
I think it is reasonable to change the ```.ipynb``` used for the test, so that it does not produce a latex error.

Reply:
+1

Reply:
The error occurs in [this notebook](https://github.com/jupyter/nbconvert/blob/master/nbconvert/exporters/tests/files/notebook2.ipynb). The error is in the last markdown cell.

Change:
```
$$$$
\[\]
$$
```
to
```
$$ $$
\[ \]
$ $
```
and the conversion will be successful. Pandoc butchers the empty math blocks without the space in the middle.